File: bitget-java-sdk-api/src/main/java/com/bitget/openapi/api/spot/SpotPublicApi.java
package com.bitget.openapi.api.spot;

import com.bitget.openapi.dto.response.ResponseResult;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * @Author: bitget-sdk-team
 * @Date: 2022-09-30 10:46
 * @DES: spot order api
 */
public interface SpotPublicApi {

    /**
     * Get server time
     * @return ResponseResult
     */
    @GET("/api/spot/v1/public/time")
    Call<ResponseResult> time();

    /**
     * Basic information of currency
     * @return ResponseResult
     */
    @GET("/api/spot/v1/public/currencies")
    Call<ResponseResult> currencies();

    /**
     * Get all product information
     * @return ResponseResult
     */
    @GET("/api/spot/v1/public/products")
    Call<ResponseResult> products();

    /**
     * Get single product information
     * @param symbol
     * @return ResponseResult
     */
    @GET("/api/spot/v1/public/product")
    Call<ResponseResult> product(@Query("symbol") String symbol);

}

File: bitget-php-sdk-api/src/api/spot/SpotPublicApi.php
<?php


namespace bitget\api\spot;


use bitget\internal\BitgetApiClient;

class SpotPublicApi
{
    const BASE_URL = "/api/spot/v1/public";

    var BitgetApiClient $BitgetApiClient;

    public function __construct($BitgetApiClient)
    {
        $this->BitgetApiClient = $BitgetApiClient;
    }

    /**
     * Get server time
     * @return string
     */
    public function time():string
    {
        return $this->BitgetApiClient->doGet(self::BASE_URL . "/time", null);
    }

    /**
     * Basic information of currency
     * @return string
     */
    public function currencies():string
    {
        return $this->BitgetApiClient->doGet(self::BASE_URL . "/currencies", null);
    }

    /**
     * Get all product information
     * @return string
     */
    public function products():string
    {
        return $this->BitgetApiClient->doGet(self::BASE_URL . "/products", null);
    }

    /**
     * Get single product information
     * @param $symbol
     * @return string
     */
    public function product(string $symbol):string
    {
        $params = array("symbol" => $symbol);
        return $this->BitgetApiClient->doGet(self::BASE_URL . "/product", $params);
    }

}